### PR TITLE
Remove bad \grammarterm for "injected-class-name"

### PR DESCRIPTION
**Pull request by timsong-cpp**

### CODE REVIEW

**Comment by tkoeppe** [comment]
Why so many commits -- squash and rebase?

**Comment by timsong-cpp** [comment]
Figured separate commits might be easier to review, but OK, squashed.

File: source/statements.tex [hunk 1]
```diff
@@ -672,8 +672,8 @@
 class type \tcode{C}, the \grammarterm{unqualified-id}{s}
 \tcode{begin} and \tcode{end} are looked up in the scope of \tcode{C}
 as if by class member access lookup~(\ref{basic.lookup.classref}), and if either
-(or both) finds at least one declaration, \grammarterm{begin-expr} and
-\grammarterm{end-expr} are \tcode{__range.begin()} and \tcode{__range.end()},
+(or both) finds at least one declaration, \textit{begin-expr} and
```
Comment (tkoeppe) [inline source/statements.tex]: We don't really want to introduce *new* `\textit`s. Consult with @jensmaurer for details.

Comment (timsong-cpp) [inline source/statements.tex]: Every other occurrence of *meow-expr* in this very section uses `\textit`, which is why I used them here.

Comment (tkoeppe) [inline source/statements.tex]: I understand that. We still need to talk about this.

Comment (jensmaurer) [inline source/statements.tex]: It seems to me _begin-expr_ should be \placeholder, actually.

Comment (timsong-cpp) [inline source/statements.tex]: So `\placeholder` it throughout? That was admittedly my first thought, before seeing all-`\textit`s in the surroundings.

Comment (jensmaurer) [inline source/statements.tex]: See #1558.

File: source/declarators.tex [hunk 2]
```diff
@@ -2110,7 +2110,7 @@
     attribute-specifier-seq\opt decl-specifier-seq\opt declarator virt-specifier-seq\opt{} \terminal{ = default ;}
 \end{ncbnf}
 
-is called an \grammarterm{explicitly-defaulted} definition.
+is called an \term{explicitly-defaulted} definition.
 A function that is explicitly defaulted shall
```
Comment (jensmaurer) [inline source/declarators.tex]: This should be `\defn{explicitly-defaulted definition}`, probably with "definition!explicitly-defaulted" as the main index entry and a "see" pointer for "explicitly-defaulted definition".

Comment (timsong-cpp) [inline source/declarators.tex]: There's already a `\indextext{definition!function!explicitly-defaulted}` ~7 lines above, which is why I went with `\term`.

Comment (jensmaurer) [inline source/declarators.tex]: In my opinion, \term should only (if at all) survive for stuff that is local enough not to ever be referenced from elsewhere, and thus not needing an index entry at all. For this particular case, it seems the \indextext from before should simply be folded into the \defnx to be used after "is called an".

I'm slightly concerned that we seem to define "explicitly-defaulted definition" as the technical term, although we mean "explicitly-defaulted function definition". That probably needs a bit of research what we actually want here.

**Comment by jensmaurer** [comment]
This needs a rebase.

**Comment by timsong-cpp** [comment]
I dropped the explicitly-defaulted change; the indexing issue is above my pay grade. The only remaining change is injected-class-name.

**Comment by tkoeppe** [comment]
@zygoloid: What do you think? It looks like the PR in its present form improves consistency.

**Comment by jensmaurer** [comment]
Yes, it removes a bad \grammarterm.

**Pull request merged**